lists-service/cicd/deploy/pipeline/pipeline.yaml

```diff
@@ -233,26 +242,50 @@ Resources:
-                  { "name":"SUB_DOMAIN", "value":"#{ExportConfigNS.SUB_DOMAIN}" },
-                  { "name":"USERDETAILS_BASE_URL", "value":"#{ExportConfigNS.USERDETAILS_BASE_URL}" }
-                ]
+              EnvironmentVariables:
+                Fn::Sub:
+                  - |
+                    [
+                      { "name":"AWS_ACCOUNT_ID", "value":"${AWS::AccountId}" },
+                      { "name":"BIE_BASE_URL", "value":"#{ExportConfigNS.BIE_BASE_URL}" },
+                      { "name":"CERTIFICATE_ARN", "value":"#{ExportConfigNS.CERTIFICATE_ARN}" },
+                      { "name":"ECR_REPO", "value":"${EksRepo}" },
+                      { "name":"EKS_CLUSTER_NAME", "value":"${EksClusterName}" },
+                      { "name":"HELM_RELEASE_NAME", "value":"#{ExportConfigNS.HELM_RELEASE_NAME}" },
+                      { "name":"HOSTED_ZONE", "value":"#{ExportConfigNS.HOSTED_ZONE}" },
+                      { "name":"IMAGE_TAG", "value":"#{ExportConfigNS.IMAGE_TAG}" },
+                      { "name":"IMAGES_BASE_URL", "value":"#{ExportConfigNS.IMAGES_BASE_URL}" },
+                      { "name":"LISTS_SECRET_NAME", "value":"${ListSecretName}" },
+                      { "name":"MONGODB_HOST", "value":"${MongoDbHost}" },
+                      { "name":"MONGODB_USERNAME", "value":"#{ExportConfigNS.ADMIN_USER}" },
+                      { "name":"NAMEMATCHING_BASE_URL", "value":"#{ExportConfigNS.NAMEMATCHING_BASE_URL}" },
+                      { "name":"OIDC_CLIENT_ID", "value":"${AppClientId}" },
+                      { "name":"OIDC_DISCOVERY_URI", "value":"${OidcDiscoveryUrl}" },
+                      { "name":"SUB_DOMAIN", "value":"#{ExportConfigNS.SUB_DOMAIN}" },
+                      { "name":"USERDETAILS_BASE_URL", "value":"#{ExportConfigNS.USERDETAILS_BASE_URL}" }
+                    ]
+                  - AppClientId:
+                      Fn::ImportValue:
+                        Fn::Sub: "${pBaseStackName}-AppClientId"
+                    EksClusterName:
+                      Fn::ImportValue:
+                        Fn::Sub: "${pRegolithStackName}-ClusterName"
+                    EksRepo:
+                      Fn::ImportValue:
+                        Fn::Sub: "${pBaseStackName}-ListsRepositoryUri"
+                    ListSecretName:
+                      Fn::ImportValue:
+                        Fn::Sub: "${pBaseStackName}-ListsSecret"
+                    MongoDbHost:
+                      Fn::ImportValue:
+                        Fn::Sub: "${pDatabaseStackName}-DocDbClusterEndpoint"
+                    OidcDiscoveryUrl:
+                        Fn::Sub:
+                          - "https://cognito-idp.${AWS::Region}.amazonaws.com/${UserPoolId}/.well-known/openid-configuration"
+                          - UserPoolId:
+                              Fn::ImportValue:
+                                Fn::Sub: "${pBaseStackName}-UserPoolId"
+
```

Comment:
Oh cool, that looks like a good way to include imports as environment vars in the JSON. Is it possible to retain the function shortform for !Sub vs Fn::Sub at the beginning?

Reply:
It may be possible use `!Sub` however there was a warning in the AWS doco regarding using `!Sub` wishing `!ImportValue` and I was having issues with getting it work work so I reverted to long form for everything. 

https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/intrinsic-function-reference-importvalue.html
> You can't use the short form of !ImportValue when it contains the short form of !Sub.